### PR TITLE
New location for the contributing docs

### LINKED ISSUE #2880: Refactor documentation for 'documentation contributions'
## Description

As a user of the TRE project, I want to know how I can feedback on the documentation so that it can be improved.
Currently, the 'How to Contribute to our Documentation' section of text is located at the end of the 'Terms and Definition' section of the overview, it should be moved to a more prominent place.

## Acceptance criteria

- [ ] The 'How to Contribute to our Documentation' is present in the document navigation tree on the left-hand side of the screen.

### PR DESCRIPTION
**Pull request by itye-msft**
# Resolves #2880 

## What is being addressed

Correcting the location of the contributing page content.

## How is this addressed
Based on other OSS projects which I reviewed as preparation - 

This PR creates a new page at the root of the mkdocs called "Contributing"

And it combines both:

[Terms and Definitions - Azure TRE (microsoft.github.io)](https://microsoft.github.io/AzureTRE/using-tre/terms-definitions/#how-to-contribute-to-our-documentation)  - (how to contribute to the docs)

and

[Projects · AzureTRE (github.com)](https://github.com/microsoft/AzureTRE/blob/main/CONTRIBUTING.md) (the code contributing page).

 
On all other OSS projects I have seen, those are combined in the mkdocs on a single page, in a very readable manner, and the old contributing.md contains a pointer to the new doc page.
e.g [minikube/CONTRIBUTING.md at master · kubernetes/minikube (github.com)](https://github.com/kubernetes/minikube/blob/master/CONTRIBUTING.md)
or [fastapi/CONTRIBUTING.md at master · tiangolo/fastapi (github.com)](https://github.com/tiangolo/fastapi/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# How to contribute to Azure TRE
+
+First of all thank you. Azure TRE is a collective effort to make it easier to create and manage trusted research environment.
```
Comment (marrobi) [inline docs/contributing.md:3]: Not sure need this sentence? Maybe would be useful to have a link to the developer docs, as we now have them.

File: docs/contributing.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+* Open a new GitHub pull request with the patch.
+* Ensure the PR description clearly describes the problem and solution. Include the relevant issue number if applicable.
+* Before submitting, please double check on the current coding conventions used.
+
+## Do you intend to add a new feature or change an existing one?
+
+* Open a [new feature issue](https://github.com/microsoft/AzureTRE/issues/new?assignees=&labels=feature&template=feature_request.md&title=) so we can first discuss the changes there. After the discussion you may send a PR with relevant changes. Do not send a PR without discussion on an issue.
+
+## Do you want to contribute to Azure TRE?
```
Comment (marrobi) [inline docs/contributing.md:20]: This feels like it's the title of this doc? Is this section needed or is it covered above?

**Review by marrobi — APPROVED** [review]
Couple of suggestions, improving on what's there. Thanks.

File: docs/contributing.md (hunk 2 above)
Comment (itye-msft) [reply docs/contributing.md]: Right. This is actually redundant and I removed it

File: docs/contributing.md (hunk 1 above)
Comment (marrobi) [reply docs/contributing.md]: If look at contributing guides for other OSS projects they typically describe how to clone the repo etc.

Maybe need to link to https://microsoft.github.io/AzureTRE/using-tre/local-development/local-development/

Although is a bit confusing that we have a contributing section visible here too: https://microsoft.github.io/AzureTRE/tre-developers/api/

**Review by tamirkamara — APPROVED** [review]

**Pull request merged**